src/robot/libraries/String.py

```diff
@@ -237,29 +237,38 @@ def get_lines_matching_pattern(self, string, pattern, case_insensitive=False):
             matches = lambda line: fnmatchcase(line, pattern)
         return self._get_matching_lines(string, matches)
 
-    def get_lines_matching_regexp(self, string, pattern):
+    def get_lines_matching_regexp(self, string, pattern, exact_match=True):
         """Returns lines of the given ``string`` that match the regexp ``pattern``.
 
         See `BuiltIn.Should Match Regexp` for more information about
         Python regular expression syntax in general and how to use it
-        in Robot Framework test data in particular. A line matches
-        only if it matches the ``pattern`` fully. Notice that to make
-        the match case-insensitive, you need to embed case-insensitive
-        flag into the pattern.
-
-        Lines are returned as one string catenated back together with
+        in Robot Framework test data in particular. By default ``exact_match``
+        is True causing line matches only if it matches the ``pattern`` fully.
+        When passing ``exact_match`` as False, this keyword uses the matching
```

Comment:
`exact_match` doesn't need to be exactly `True` or `False` after we recently introduced `is_truthy` that you also use here. Take a look at other `get_lines_xxx` keywords for examples how to explain that.